python/tk_multi_publish2/base_hooks/publish_plugin.py

```diff
@@ -451,15 +451,18 @@ def create_settings_widget(self, parent):
 
     def get_ui_settings(self, widget):
         """
-        Invoked by the publisher when the selection changes so the new settings
-        can be applied on the previously selected tasks.
+        Invoked by the publisher when the selection changes. It is performed on the
+        task that was just previously selected so that the publisher can gather the settings.
+        These settings can then be later used to repopulate the custom UI if the task
+        gets selected again, and also be used in the publishing process by the plugin.
```

Comment:
I think it's fine as-is, so you can ignore me, but this is what I would write (just slightly cleaned up):

Invoked by the Publisher when the selection changes. This method gathers the settings on the previously selected task, so that they can be later used to repopulate the custom UI if the task gets selected again, and also be used by the plugin.